lib/run_loop/physical_device/life_cycle.rb

```diff
@@ -0,0 +1,260 @@
+module RunLoop
+  # @!visibility private
+  module PhysicalDevice
+
+    # Raised when installation fails.
+    class InstallError < RuntimeError; end
+
+    # Raised when uninstall fails.
+    class UninstallError < RuntimeError; end
+
+    # Controls the behavior of various life cycle commands.
+    #
+    # You can override these values if they do not work in your environment.
+    #
+    # For cucumber users, the best place to override would be in your
+    # features/support/env.rb.
+    #
+    # For example:
+    #
+    # RunLoop::PhysicalDevice::LifeCycle::DEFAULT_OPTIONS[:timeout] = 60
+    DEFAULT_OPTIONS = {
+      :install_timeout => RunLoop::Environment.ci? ? 120 : 30
+    }
+
+    # @!visibility private
+    class LifeCycle
+
+      require "run_loop/abstract"
+      include RunLoop::Abstract
+
+      require "run_loop/shell"
+      include RunLoop::Shell
+
+      attr_reader :device
+
+      def initialize(device)
+        if !device.physical_device?
+          raise ArgumentError, %Q[Device:
+
+#{device}
+
+must be a physical device.]
```

Comment:
There is already an existing API for simulators in CoreSimulator.  The branch on sim or device will happen further up the stack.